heron/tmaster/tests/cpp/server/stateful_checkpointer_unittest.cpp

```diff
@@ -0,0 +1,241 @@
+/*
+ * Copyright 2015 Twitter, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <map>
+#include <string>
+#include <sstream>
+#include <thread>
+#include <vector>
+#include "gtest/gtest.h"
+#include "proto/messages.h"
+#include "basics/basics.h"
+#include "errors/errors.h"
+#include "threads/threads.h"
+#include "network/network.h"
+#include "basics/modinit.h"
+#include "errors/modinit.h"
+#include "threads/modinit.h"
+#include "network/modinit.h"
+#include "config/topology-config-vars.h"
+#include "config/topology-config-helper.h"
+#include "config/physical-plan-helper.h"
+#include "manager/stateful-checkpointer.h"
+
+// Generate a dummy topology
+static heron::proto::api::Topology* GenerateDummyTopology(
```

Comment:
I've seen this method in a number of test classes. If you have ideas for how to effectively share test code like this that would be awesome.